@here/harp-mapview/lib/ClipPlanesEvaluator.ts

```diff
@@ -0,0 +1,524 @@
+     * @param camera
+     * @param projection
+     */
+    protected getCameraAltitude(camera: THREE.Camera, projection: Projection) {
+        return projection.groundDistance(camera.position);
+    }
+
+    private evaluateDistancePlanarProj(
+        camera: THREE.Camera,
+        projection: Projection
+    ): { near: number; far: number } {
+        assert(projection.type !== ProjectionType.Spherical);
+
+        let nearPlane: number = this.nearMin;
+        let farPlane: number = this.nearMin + this.planesMaxDistance;
+
+        // Calculate distance to closest point on the ground.
+        const groundDistance = this.getCameraAltitude(camera, projection);
+        // Acquire camera forward (look at) vector (store in tmpVectors[0]).
+        const lookAt = camera.getWorldDirection(this.m_tmpVectors[0]);
+        const tiltCos = Math.cos(
+            lookAt.angleTo(AltitudeBasedClipPlanesEvaluator.GROUND_NORMAL_INV_PLANAR_PROJ)
+        );
+        // At least try to keep margins along the eye vector (center of the view) in tact with
+        // pitch angle changes - this does not solve all tilt change problems, rather use
+        // more sophisticated evaluator.
+        nearPlane = groundDistance - this.nearMargin / tiltCos;
+        farPlane = groundDistance + this.farMargin / tiltCos;
+
+        // Apply the constraints.
+        nearPlane = Math.max(nearPlane, this.nearMin);
+        farPlane = Math.min(farPlane, nearPlane + this.planesMaxDistance);
+
+        return { near: nearPlane, far: farPlane };
+    }
+
+    private evaluateDistanceSphericalProj(
+        camera: THREE.Camera,
+        projection: Projection
+    ): { near: number; far: number } {
+        assert(projection.type === ProjectionType.Spherical);
+
+        let nearPlane: number = this.nearMin;
+        let farPlane: number = this.nearMin + this.planesMaxDistance;
+
+        // This solution computes near and far plane for a set up where
+        // the camera is looking at the center of the scene.
+
+        //         , - ~ ~ ~ - ,
+        //     , '               ' ,
+        //   ,                       ,
+        //  ,                         ,
+        // ,                           ,
+        // ,             . O           ,
+        // ,             | .           ,
+        //  ,            |   .  r     ,
+        //   ,           |      .    ,
+        //     ,         |        , '_____ far
+        //       ' -_, _ | _ ,  ' / T
+        //     near      |      /
+        //               |    / t
+        //             d |  /
+        //               |/
+        //               C
+
+        // The near plane calculus is quite straight forward and works the same as for planar
+        // projections. We simply search for the closest point of the ground just above
+        // the camera, then we apply margin to it along the sphere surface normal:
+        nearPlane = this.getCameraAltitude(camera, projection) - this.nearMargin;
```

Comment:
Yes and no... it does not evaluate extrusion height (this will be the part of third ticket regarding evaluators), but somehow it may, because you may simply set the margin high enough to cover (in frustum) all tallest buildings in the world.